pkg/cmd/release/git.go

```diff
@@ -249,25 +250,39 @@ func findHealthyBuild(potentialRefs []string) (buildInfo, error) {
 	return buildInfo{}, fmt.Errorf("no ref found")
 }
 
-// remoteBranchExists checks if a branch exists in a remote repository, assuming the remote name is `origin`.
-func remoteBranchExists(branch string) (bool, error) {
-	cmd := exec.Command("git", "ls-remote", "--refs", "origin", "refs/heads/"+branch)
+// listRemoteBranches retrieves a list of remote branches using a pattern, assuming the remote name is `origin`.
+func listRemoteBranches(pattern string) ([]string, error) {
+	cmd := exec.Command("git", "ls-remote", "--refs", "origin", "refs/heads/"+pattern)
```

Comment:
with the logic for `generateRepoList` getting more complex, it would be great to add a test for `generateRepoList` somehow (e.g. given`releasedVersion`, and maybe/not a `nextVersion`, branches given `listRemoteBranches` is X)

but I can't think of a way to how to do this with `` calling `git ls-remote ...` directly 🤔 .

Reply:
I have been thinking about how to test these. One of the options would be creating local temporary repos with some commits/contents we can test against.